Clarify environment variables and assetPath function in assets reference (#2592)
sensu/sensu-docs: content/sensu-go/5.21/reference/assets.md

For further examples of Sensu users who have added the ability to use a community plugin as an asset, see [this Discourse post](https://discourse.sensu.io/t/how-to-use-the-sensu-plugins-kubernetes-plugin/1286).
{{% /notice %}}

### Default cache directory

system  | sensu-backend                         | sensu-agent
--------|---------------------------------------|-------------
Linux   | `/var/cache/sensu/sensu-backend`      | `/var/cache/sensu/sensu-agent`
Windows | N/A                                   | `C:\ProgramData\sensu\cache\sensu-agent`

If the requested asset is not in the local cache, it is downloaded from the asset URL.

> The Sensu backend does not currently provide any storage for assets.

Reviewer: ```suggestion
The Sensu backend acts as an index of asset builds, and does not provide storage or hosting for the build artifacts.
```

Sensu expects assets to be retrieved over HTTP or HTTPS.

### Example asset structure

{{< code shell >}}
sensu-example-handler_1.0.0_linux_amd64
├── CHANGELOG.md
├── LICENSE
├── README.md
└── bin
  └── my-check.sh
└── lib
└── include
{{< /code >}}

## Asset path

When you download and install an asset, the asset's local path on disk is exposed to asset consumers via environment variables.
This allows you to retrieve the asset's path as an [environment variable][14] from a check, handler, hook, or mutator or using a [custom function for token substitution][17], `assetPath`, from a check or hook.

### Environment variable for asset path

> If you included a configuration file in the `include` directory of the asset [`sensu-plugins-windows`][4], you can use the asset path environment variable to reference the asset from your check, handler, hook, or mutator:

Reviewer: ```suggestion
For each runtime asset, a corresponding environment variable will be available in the `command` context.
Sensu generates the environment variable name by capitalizing the asset name, replacing any special characters with underscores, and appending the `_PATH` suffix.
The value of the variable will be the path on disk where the asset build has been unpacked.
```

I think the example usecase "If you included a configuration file in the `include` directory of the asset..." belongs in a different section where we describe "why you might want this".

`$SENSU_PLUGINS_WINDOWS_PATH/include/config.yaml`

{{% notice note %}}

> **NOTE**: Sensu always appends the suffix `_PATH` for environment variables that are automatically generated for asset paths.

Reviewer: ```suggestion
```

{{% /notice %}}

### Token substitution for asset path

> The asset path includes the asset's checksum, which changes every time the asset is updated.

Reviewer: Most of these words belongs in "why you want this" section. Other salient points we need to cover:

* Most of the time, you won't need to know the path to the asset files on disk
* Sometimes you need this, like to provide an argument to a command.
* Both the environment variable and the token substitution help you get this information. 
* You can use the environment variable in every context, but the token substitution only works in check or hook commands.

This would normally require you to manually update the commands for any of your checks or hooks that consume the asset.
However, the `assetPath` token subsitution function allows you to substitute the asset's local path on disk, so you will not need to manually update your check or hook commands every time the asset is updated.

{{% notice note %}}

> **NOTE**: The `assetPath` function is only available where token substitution is available: checks and hooks.

Reviewer: ```suggestion
**NOTE**: The `assetPath` function is only available where token substitution is available: the `command` attribute of a check or hook resource.
```

For further examples of Sensu users who have added the ability to use a community plugin as an asset, see [this Discourse post](https://discourse.sensu.io/t/how-to-use-the-sensu-plugins-kubernetes-plugin/1286).
{{% /notice %}}

### Default cache directory

system  | sensu-backend                         | sensu-agent
--------|---------------------------------------|-------------
Linux   | `/var/cache/sensu/sensu-backend`      | `/var/cache/sensu/sensu-agent`
Windows | N/A                                   | `C:\ProgramData\sensu\cache\sensu-agent`

If the requested asset is not in the local cache, it is downloaded from the asset URL.
The Sensu backend acts as an index of asset builds, and does not provide storage or hosting for the build artifacts.
Sensu expects assets to be retrieved over HTTP or HTTPS.

### Example asset structure

{{< code shell >}}
sensu-example-handler_1.0.0_linux_amd64
├── CHANGELOG.md
├── LICENSE
├── README.md
└── bin
  └── my-check.sh
└── lib
└── include
{{< /code >}}

## Asset path

When you download and install an asset, the asset files are saved to a local path on disk.

> Most of the time, you won't need to know this path &mdash; except when you want to provide an argument to a command that uses the asset.

Reviewer: ```suggestion
Most of the time, you won't need to know this path &mdash; except in cases where you need to provide the full path to asset files as part of a command argument.
```

> In that case, you *will* need to know the full explicit path to the asset files.

Reviewer: ```suggestion
```

> The asset path includes the asset's checksum, which changes every time the asset is updated.

Reviewer: ```suggestion
The asset directory path includes the asset's checksum, which changes every time underlying asset artifacts are updated.
```

This would normally require you to manually update the commands for any of your checks, handlers, hooks, or mutators that consume the asset.

> However, the local path on disk is exposed to asset consumers via [environment variables][14] and the [`assetPath` custom function for token substitution][17].

Author: ```suggestion
However, the local path on disk is exposed to asset consumers via [environment variables][14] and the [`assetPath` custom function for token substitution][17], which allows you to avoid manual updates.
```

> However, the local path on disk is exposed to asset consumers via [environment variables][14] and the [`assetPath` custom function for token substitution][17], which allows you to avoid manual updates.

Reviewer: ```suggestion
However, because the asset directory path is exposed to asset consumers via [environment variables][14] and the [`assetPath` custom function for token substitution][17], you can avoid these manual updates.
```

> You can retrieve the asset's path as an environment variable in every context (in other words, for a check, handler, hook, or mutator).

Author: ```suggestion
You can retrieve the asset's path as an environment variable in the `command` context for checks, handlers, hooks, and mutators.
```
